mesonbuild/modules/python.py

```diff
@@ -286,12 +291,17 @@ def __init__(self, interpreter, python, info):
         self.interpreter = interpreter
         self.subproject = self.interpreter.subproject
         prefix = self.interpreter.environment.coredata.get_builtin_option('prefix')
+        self.is_debug = self.interpreter.environment.coredata.get_builtin_option('buildtype') == 'debug'
         self.variables = info['variables']
         self.paths = info['paths']
         install_paths = info['install_paths']
         self.platlib_install_path = os.path.join(prefix, install_paths['platlib'][1:])
         self.purelib_install_path = os.path.join(prefix, install_paths['purelib'][1:])
         self.version = info['version']
+        if mesonlib.version_compare(self.version, '>= 3.0'):
```

Comment:
that means `PythonDependency` can now use `self.major_version = python_holder.major_version`, slightly less duplication :)